course/assignments/reproducible-research-assignment.md

```diff
@@ -0,0 +1,21 @@
+## Reproducible Research Plan:
+
+Based on the [Data Reuse Plan](https://github.com/mozillascience/working-open-workshop/blob/gh-pages/handouts/data_reuse_plan_template.md) by Mozilla WOW.  
+
+Create a README.Rmd that describes the data or software package. It can be pretty sparse for now but you will populate it as you create your project. It must have the following sections (If a section does not apply to your project, remove it and when you submit your assignment, include a Notes.txt explaining why it is not applicable)
```

Comment:
I agree that the students should follow an already developed style guideline, and I think it is important that we are consistent across the lectures as well.

I think it is trickier to agree on _which_ style to follow, since R does not have official guidelines. I believe Google's recommendations appear to be sound in large, with the exception of using periods in object/variable and column names. This is confusion for me coming from and object oriented language and I would prefer snake case (a.k.a. underscores: `variable_name`) in variable names and Pascal / upper camel case in column names (`ColumnName`). But, and I keep saying this, I'm happy to conform to other standards based on advice from the more experienced R-users among us and discussion with the rest of the group.

However, I do believe underscores would be helpful for students if they are ever to read or write code written in another language, where periods have special meaning. This coding style is not without precedence in the R community, e.g. Hadley Wickham's [stat 405 course](http://stat405.had.co.nz/r-style.html) recommends underscores for variable names, as does [this R style guide](https://csgillespie.wordpress.com/2010/11/23/r-style-guide/). The most compelling reason to use underscores in our case might be to stay consistent with [the recommendations in the tidyverse guidelines](http://style.tidyverse.org/index.html) (naturally similar to those from stat 405), since we will largely be using packages from the tidyverse and it would be confusing to mix these conventions with those from another style guide. The tidyverse favors underscores for objects, functions, and parameters, e.g. `tbl_df`, `group_by`, `geom_point`, `facet_wrap`, etc. The tidyverse style guide is the most exhaustive of the ones I have seen, and often includes justifications of the chosen style, which helps understanding and memorization. There is also a package, [lintr](https://github.com/jimhester/lintr), for automatically controlling for adherence to the tidyverse guidelines, and I believe this is the syntax checker that is builtin to Rstudio. 

Some additional info: Bioconductor recommends lower camel case. [Some Swedish research](http://journal.r-project.org/archive/2012-2/RJournal_2012-2_Baaaath.pdf) shows that all three styles are common in existing CRAN packages. [This is a helpful SO question](https://stackoverflow.com/questions/1944910/what-is-your-preferred-style-for-naming-variables-in-r
), with several good answers.